compiler/noirc_frontend/src/hir/def_collector/dc_mod.rs

```diff
@@ -241,20 +252,38 @@ impl<'a> ModCollector<'a> {
                                 where_clause,
                                 return_type,
                             ));
+                            func_ids_in_trait.insert(func_id);
                             unresolved_functions.push_fn(self.module_id, func_id, impl_method);
                         }
                         None => {
-                            let error = DefCollectorErrorKind::TraitMissedMethodImplementation {
+                            let error = DefCollectorErrorKind::TraitMissingMethod {
                                 trait_name: trait_def.name.clone(),
                                 method_name: name.clone(),
                                 trait_impl_span: trait_impl.object_type_span,
                             };
                             errors.push(error.into_file_diagnostic(self.file_id));
                         }
                     }
+                } else {
+                    for (_, func_id, _) in &matching_fns {
+                        func_ids_in_trait.insert(*func_id);
+                    }
```

Comment:
There's a comment for checking `matching_fns.len() > 1 => duplicate definition`, but I don't see the check here

Reply:
In this case, the Duplicate error is thrown from inside `collect_functions`, we don't do the check here since we'd duplicate Duplicate errors.
 https://github.com/noir-lang/noir/pull/2652/commits/3cc2a5ef7a5ad8dfad30fdf7646560f3a13ee53f